src/main/java/gyro/aws/ec2/PeeringConnectionResource.java

```diff
@@ -301,16 +301,73 @@ private VpcPeeringConnection getVpcPeeringConnection(Ec2Client client) {
+        }
+        if (getAllowEgressFromLocalVpcToRemoteClassicLink()) {
+            req.allowEgressFromLocalVpcToRemoteClassicLink(getAllowEgressFromLocalVpcToRemoteClassicLink());
+        }
+
         client.modifyVpcPeeringConnectionOptions(r -> r.vpcPeeringConnectionId(getId())
-            .accepterPeeringConnectionOptions(acp -> acp.allowDnsResolutionFromRemoteVpc(
-                getPeerAllowDnsResolutionFromRemoteVpc())
-                .allowEgressFromLocalClassicLinkToRemoteVpc(getPeerAllowEgressFromLocalClassicLinkToRemoteVpc())
-                .allowEgressFromLocalVpcToRemoteClassicLink(getPeerAllowEgressFromLocalVpcToRemoteClassicLink()))
-            .requesterPeeringConnectionOptions(req -> req.allowDnsResolutionFromRemoteVpc(
-                getAllowDnsResolutionFromRemoteVpc())
-                .allowEgressFromLocalClassicLinkToRemoteVpc(getAllowEgressFromLocalClassicLinkToRemoteVpc())
-                .allowEgressFromLocalVpcToRemoteClassicLink(getAllowEgressFromLocalVpcToRemoteClassicLink()))
-        );
+            .requesterPeeringConnectionOptions(req.build()));
+    }
+
+    private void modifyAccepterPeeringConnectionSettings(Ec2Client client) {
+        PeeringConnectionOptionsRequest.Builder acp = PeeringConnectionOptionsRequest.builder();
+
+        if (getPeerAllowDnsResolutionFromRemoteVpc()) {
+            acp.allowDnsResolutionFromRemoteVpc(getPeerAllowDnsResolutionFromRemoteVpc());
+        }
+        if (getPeerAllowEgressFromLocalClassicLinkToRemoteVpc()) {
+            acp.allowEgressFromLocalClassicLinkToRemoteVpc(getPeerAllowEgressFromLocalClassicLinkToRemoteVpc());
+        }
+        if (getPeerAllowEgressFromLocalVpcToRemoteClassicLink()) {
+            acp.allowEgressFromLocalVpcToRemoteClassicLink(getPeerAllowEgressFromLocalVpcToRemoteClassicLink());
+        }
+
+        client.modifyVpcPeeringConnectionOptions(r -> r.vpcPeeringConnectionId(getId())
+            .accepterPeeringConnectionOptions(acp.build()));
+    }
+
+    private void modifyPeeringConnectionSettings(Ec2Client client) {
+        if (!getVpc().getRegion().equals(getPeerVpc().getRegion())) {
+            try (Ec2Client accepterClient = createClient(Ec2Client.class, getPeerVpc().getRegion(), "")) {
+                accepterClient.acceptVpcPeeringConnection(
+                    r -> r.vpcPeeringConnectionId(getId()).overrideConfiguration()
+                );
+                waitForStatus(client, "active");
+                modifyRequesterPeeringConnectionSettings(client);
+                modifyAccepterPeeringConnectionSettings(accepterClient);
+            }
+        } else {
+            client.acceptVpcPeeringConnection(r -> r.vpcPeeringConnectionId(getId()));
+            waitForStatus(client, "active");
+            modifyRequesterPeeringConnectionSettings(client);
+            modifyAccepterPeeringConnectionSettings(client);
+        }
+    }
```

Comment:
We want to pass whether this is a cross region connection into the modify calls, as some fields cannot be adjusted for cross region peering connections

Also, we only want to make the `acceptVpcPeeringConnection` call if this is a create operation.

```suggestion
    private void modifyPeeringConnectionSettings(Ec2Client requesterClient, boolean create) {
        boolean crossRegion = !getVpc().getRegion().equals(getPeerVpc().getRegion());
        Ec2Client accepterClient =
            crossRegion ? createClient(Ec2Client.class, getPeerVpc().getRegion(), "") : requesterClient;

        if (create) {
            accepterClient.acceptVpcPeeringConnection(r -> r.vpcPeeringConnectionId(getId()));
            waitForStatus(requesterClient, VpcPeeringConnectionStateReasonCode.ACTIVE);
        }

        modifyRequesterPeeringConnectionSettings(requesterClient, crossRegion);
        modifyAccepterPeeringConnectionSettings(accepterClient, crossRegion);
    }
```